Source/Widgets/CabbageImage.cpp

```diff
@@ -155,21 +157,19 @@ void CabbageImage::valueTreePropertyChanged (ValueTree& valueTree, const Identif
 
 void CabbageImage::updateImage(ValueTree& valueTree)
 {
-	String fileBase64 = CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::file);
-
-	if (fileBase64.isNotEmpty()) {
-		MemoryOutputStream out;
-		bool result = Base64::convertFromBase64(out, fileBase64);
-		if (result)
-		{
-			img = ImageFileFormat::loadFrom(out.getData(), out.getDataSize());
-		}
-		else
-		{
-			imgFile = File(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::csdfile)).getParentDirectory().getChildFile(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::file));
-			if (File(imgFile).existsAsFile())
-				img = ImageFileFormat::loadFrom(imgFile);
-		}
+    imgBase64 = CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::file);
+    if(imgBase64.isNotEmpty())
+    {
+        MemoryOutputStream out;
+        bool result = Base64::convertFromBase64(out, imgBase64);
+        if(result)
+            img = ImageFileFormat::loadFrom(out.getData(), out.getDataSize());
+    }
+    else 
+    {
+		imgFile = File(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::csdfile)).getParentDirectory().getChildFile(CabbageWidgetData::getStringProp(valueTree, CabbageIdentifierIds::file));

```

Comment:
This block will only be hit if the file() identifier returns an empty string, in which case it makes no sense in trying to load an image. We need to first check if the file string is not empty, and then proceed to validate it as either a base64 string, or a file.